diff -ur soundtouch.orig/include/STTypes.h soundtouch/include/STTypes.h
--- soundtouch.orig/include/STTypes.h	2009-01-25 17:32:53.000000000 +0100
+++ soundtouch/include/STTypes.h	2009-06-21 18:23:27.352532633 +0200
@@ -87,7 +87,7 @@
  
  #endif
 
-    #if (WIN32 || __i386__ || __x86_64__)
+    #if (WIN32 || __i386__)
         /// Define this to allow X86-specific assembler/intrinsic optimizations. 
         /// Notice that library contains also usual C++ versions of each of these
         /// these routines, so if you're having difficulties getting the optimized 
