--- SoundTouch-1.3.0/configure.ac.orig	2005-02-06 13:05:52.000000000 +0100
+++ SoundTouch-1.3.0/configure.ac	2005-05-03 11:41:49.870170312 +0200
@@ -35,7 +35,6 @@
 dnl ############################################################################
 dnl # Checks for programs                                                      #
 dnl ############################################################################
-CXXFLAGS= dnl AC_PROG_CXX automatically sets CXXFLAGS to -g -O2 if g++ is the C++ compiler and CXXFLAGS hasn't been set, so by setting CXXFLAGS, we can avoid this.
 AC_PROG_CXX
 #AC_PROG_AWK
 AC_PROG_CC
@@ -87,6 +86,23 @@
         AC_DEFINE(FLOAT_SAMPLES,1,[Use Float as Sample type])
 fi
 
+AC_LANG(C++)
+
+AC_MSG_CHECKING([if compiler accepts -msse])
+
+old_CXXFLAGS=$CXXFLAGS
+CXXFLAGS="-msse"
+
+AC_TRY_COMPILE(, [return 0;], [
+    CFLAG_MSSE="-msse"
+    AC_MSG_RESULT([yes])
+    ], [
+    CFLAG_MSSE=""
+    AC_MSG_RESULT([no])
+    ])
+CXXFLAGS=$old_CXXFLAGS
+
+AC_SUBST(CFLAG_MSSE)
 
 
 
--- soundtouch-1.3.1/source/SoundTouch/Makefile.am.orig	2006-03-19 13:08:36.000000000 +0100
+++ soundtouch-1.3.1/source/SoundTouch/Makefile.am	2006-05-21 19:43:24.516711000 +0200
@@ -35,8 +35,9 @@
 
 # ??? test for -fcheck-new in configure.ac
 # other compiler flags to add
-AM_CXXFLAGS=-O3 -msse -fcheck-new -I../../include
+AM_CXXFLAGS=-fcheck-new -I../../include
 
 # other linking flags to add
 #libSoundTouch_la_LIBADD=
 
+sse_optimized.lo: AM_CXXFLAGS+=$(CFLAG_MSSE)
--- SoundTouch-1.3.0/source/example/bpm/Makefile.am.orig	2003-12-27 14:02:10.000000000 +0100
+++ SoundTouch-1.3.0/source/example/bpm/Makefile.am	2005-05-03 11:43:11.554752368 +0200
@@ -33,7 +33,7 @@
 
 
 # other compiler flags to add
-AM_CXXFLAGS=-O3 -I../../../include
+AM_CXXFLAGS=-I../../../include
 
 # other linking flags to add
 #libBPM_la_LIBADD=
--- SoundTouch-1.3.0/source/example/SoundStretch/Makefile.am.orig	2003-12-27 11:42:32.000000000 +0100
+++ SoundTouch-1.3.0/source/example/SoundStretch/Makefile.am	2005-05-03 11:42:44.505864424 +0200
@@ -43,11 +43,8 @@
 ## linker. 
 soundstretch_LDADD=../../SoundTouch/libSoundTouch.la ../bpm/libBPM.la -lm
 
-## linker flags
-soundstretch_LDFLAGS=-s
-
 ## additional compiler flags
-soundstretch_CXXFLAGS=-O3 -I../bpm
+soundstretch_CXXFLAGS=-I../bpm
 
 #clean-local: 
 #	-rm -f additional-files-to-remove-on-make-clean
